src/Cake.Common/IO/FileCopier.cs

```diff
@@ -106,42 +106,45 @@ public static void CopyFiles(ICakeContext context, IEnumerable<FilePath> filePat
                 throw new DirectoryNotFoundException(message);
             }
 
-            if (preserverFolderStructure)
+            if (filePaths.Any())
```

Comment:
We only need to check if the the list is empty when `preserveFolderStructure` is `false`, so let's move this inside the `preserveFolderStructure` condition to avoid an unnecessary enumeration. Let's also invert the new `if` statement to reduce nesting (i.e. if list is empty, give up and return)

Reply:
@augustoproiete 
This is done.
Please check.

Reply:
Thanks, that's it. Cheers. Will merge after CI finishes